signac/contrib/project.py

```diff
@@ -1982,13 +1984,38 @@ def _add(_id):
         else:
             logger.debug("In-memory cache is up to date.")
 
+    # This function should be removed in signac 2.0.
+    def _migrate_internal_file(self, old_file, new_file):
+        """Migrate files based on the 1.x names to the new 2.x schema.
+
+        Various files used internally by signac are placed at the project root in
+        signac 1.x but will be placed in a .signac directory in 2.x. This function
+        provides a standardized migration protocol.
+
+        Parameters
+        ----------
+        old_file : str
+            The original filename.
+        new_file : str
+            The new filename.
+
+        """
+        old_fn = self.fn(old_file)
+        new_fn = self.fn(new_file)
+
+        if os.path.exists(old_fn):
+            _mkdir_p(os.path.dirname(new_fn))
+            os.rename(old_fn, new_fn)
+            logger.info("Moved {old_fn}->{new_fn} for signac 2.0 schema migration.")
```

Comment:
Should this be only at info level? If a user has a file moved by this function is source control, they will be really confused as to what happened.  So far we only use it on the cache and shell history, so it shouldn't be something tracked anyway.

Also, I think we want this to be
```suggestion
            logger.info(f"Moved {old_fn}->{new_fn} for signac 2.0 schema migration.")
```

Reply:
I think your concern is valid for `signac.rc`, but as you point out that is far less of a concern for the files moved in this PR. Deciding how to handle `signac.rc` migration is a much more complicated question due to concerns like what @bdice brought up in his previous comment. How long do we retain backwards compatibility? In what form? Do we make the move automatically, or require explicit migration? I figured we would need to discuss this much more in depth for `signac.rc`, which is not touched by this PR for that reason.

Good catch on the missing f-string though, that's definitely just a mistake on my part :)